manifests/init.pp

```diff
@@ -31,14 +31,31 @@
         if ! $gpgkeyfingerprint {
           fail('you must specify a `$gpgkeyfingerprint` when using `$manage_repo` on debian')
         }
+        # place the key in the keyrings directory where apt won't search for keys for all repos
+        file { '/usr/share/keyrings/lldpd.gpg':
+          source => "https://download.opensuse.org/repositories/home:/vbernat/${repourl}/Release.key",
+          owner  => 'root',
+          group  => 'root',
+          mode   => '0644',
+        }
+        # purge old key files that we installed in previous releases
+        $old_keys = ['/etc/apt/trusted.gpg.d/home_vbernat.gpg', '/etc/apt/trusted.gpg.d/home_vbernat.gpg~',]
+        $old_keys.each |Stdlib::Absolutepath $key| {
+          file { $key:
+            ensure => 'absent',
+          }
+        }
+        # previously managed by apt::key, we need to purge it from the global keyring in /etc/apt/trusted.gpg
+        include apt # required so apt::key can access variables from init.pp
+        apt::key { 'EF795E4D26E48F1D7661267B431C37A97C3E114B':
+          ensure => 'absent',
+        }
         apt::source { 'lldpd':
-          location => "http://download.opensuse.org/repositories/home:/vbernat/${repourl}",
+          location => "https://download.opensuse.org/repositories/home:/vbernat/${repourl}",
           release  => ' ',
           repos    => '/',
-          key      => {
-            id     => $gpgkeyfingerprint,
-            source => "http://download.opensuse.org/repositories/home:/vbernat/${repourl}/Release.key",
-          },
+          keyring  => '/usr/share/keyrings/lldpd.gpg',
```

Comment:
The `keyring` parameter was introduced in https://github.com/puppetlabs/puppetlabs-apt/pull/991/commits/0c35168fb0223854fb82d1a61e4769bf2297a17e so I think `metadata.json` also needs to increase the lower bound to v8.1.0.